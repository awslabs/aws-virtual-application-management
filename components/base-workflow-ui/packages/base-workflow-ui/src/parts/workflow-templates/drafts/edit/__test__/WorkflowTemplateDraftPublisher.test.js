/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowTemplateDraftPublisher from '../WorkflowTemplateDraftPublisher';

jest.mock('../../../../workflow-common/component-states/WorkflowCommonCardState');
jest.mock('../../../WorkflowTemplateCardTabs');
describe('WorkflowTemplateDraftPublisher', () => {
  it('WorkflowTemplateDraftPublisher should renders correctly', () => {
    const uiEventBus = {};
    const onCancel = jest.fn();
    const editor = { draft: { template: { propertyOverrideSummaryRows: [] } }, metaForm: {} };
    expect(
      renderer
        .create(
          <Provider>
            <WorkflowTemplateDraftPublisher.WrappedComponent
              onCancel={onCancel}
              editor={editor}
              uiEventBus={uiEventBus}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
