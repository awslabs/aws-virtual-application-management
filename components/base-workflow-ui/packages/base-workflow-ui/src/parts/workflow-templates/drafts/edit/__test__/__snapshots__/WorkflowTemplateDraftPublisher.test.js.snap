// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowTemplateDraftPublisher WorkflowTemplateDraftPublisher should renders correctly 1`] = `
Array [
  <div
    className="dimmable"
  >
    <div
      className="ui inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted text loader"
        >
          Processing
        </div>
      </div>
    </div>
    <h3
      className="ui grey header mt0 mb3"
    >
      Review & Publish
    </h3>
  </div>,
  <div>
    <button
      className="ui teal right floated button ml2"
      onClick={[Function]}
    >
      Publish
    </button>
    <button
      className="ui icon right floated left labeled button ml3"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="left arrow icon"
        onClick={[Function]}
      />
      previous
    </button>
    <button
      className="ui left floated button"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>,
]
`;
